variable/except.cpp

```diff
@@ -14,3 +15,14 @@ void throw_mismatch_error(const variable::Variable &expected,
                       to_string(actual) + '.');
 }
 } // namespace scipp::except
+
+namespace scipp::variable {
+
+std::string pretty_dtype(const Variable &var) {
+  if (!is_bins(var))
+    return to_string(var.dtype());
+  return to_string(var.dtype()) +
+         "(dtype=" + pretty_dtype(variableFactory().elem_dtype(var)) + ")";
```

Comment:
Example output:
`Cannot apply log to item dtypes VariableView(dtype=int64)`
Not quite sure this is the best solution. It might make more sense to just display the inner dtype. Or are there cases where it matters if there are bins or not?

Also (unrelated to this change), showing 'VariableView' does not really make sense. Can this be changed to something like 'bin(Variable)'?

Reply:
Yes I was wondering whether just showing the inner dtype would be enough, but it would be confusing: If someone got the error and then inspects the HTML view they would not see this printed dtype.

I agree that `VariableView` is not perfect, but I need to think a bit more about a solution. The problem is that in practice the elements behave like `Variable`, but we also have an actual `dtype=Variable`. Maybe `bin<Variable>` or something like that is really the best solution, but it is confusing since when accessed it actually behaves like `Variable`? `view<Variable>`?